Incorporate suggestions from #2578 (#2579)
modelica/ModelicaSpecification: chapters/annotations.tex

\end{lstlisting}

The mandatory \lstinline!x! and \lstinline!y! expressions are restricted to be
component references refering to a scalar variable or \lstinline!time!. If
\lstinline!x! or \lstinline!y! does not designate a variable the behavior is

> unspecified.

Reviewer: I think we should take the chance of defining this to either be an error or explicitly allowed with a defined result.

Author: Discussion continued in a suggestion.

> \lstinline!x! or \lstinline!y! does not designate a variable the behavior is
> unspecified.

Author: Something like:
```suggestion
\lstinline!x! or \lstinline!y! does not designate a scalar variable or \lstinline!time! the \lstinline!Curve! is
invalid.
```

Reviewer 2: I think this makes sense.

Reviewer: We tend to say _is an error_ rather than _is invalid_, so I think we need to also include the start of the sentence:
> It is an error if `x` or `y` does not designate a scalar variable.

Author: Discussion continued in a new suggestion.

> component references refering to a scalar variable or \lstinline!time!. If
> \lstinline!x! or \lstinline!y! does not designate a variable the behavior is
> unspecified.

Author: I couldn't figure out how to edit the suggestion to include one more row so here is a new one.
```suggestion
component references refering to a scalar variable or \lstinline!time!. It
is an error if \lstinline!x! or \lstinline!y! does not designate a scalar variable.
```
Or should we explicitly mention time again?
```suggestion
component references refering to a scalar variable or \lstinline!time!. It
is an error if \lstinline!x! or \lstinline!y! does not designate a scalar variable or \lstinline!time!.
```

Reviewer: To me it seems sufficient to consider `time` a scalar variable in this context, that is, that it doesn't need to be mentioned explicitly.

When an axis bound isn't provided, the tool computes one automatically.

The Modelica tool is responsible for showing the unit used for values at the
axis tick marks, so the axis \lstinline!label! shall not contain the unit.

If a tool does not recognize the \lstinline!unit!, it is recommended to issue a

> warning and treat the \lstinline!unit! as if it was not specified.

Reviewer: I realize that this isn't really an option as long as the `unit` is mandatory.  However, I believe that having a mandatory `Axis.unit`, which in turn causes `Plot.x` and `Plot.y` to be mandatory, is a problem that needs to be resolved separately, and that we open up a separate PR for how this should be done.

Reviewer: That separate PR is now opened as #2632.

the empty string as axis label means that no label should be shown. Variable
replacements, as described in \cref{variable-replacements}, can be used in
the \lstinline!label! of \lstinline!Axis!

\subsubsection{Plot Curves}\label{plot-curves}
The actual data to plot is specified in the \lstinline!curves! of a \lstinline!Plot!:
\begin{lstlisting}[language=modelica]
record Curve
  expression x = time "X coordinate values";
  expression y "Y coordinate values";
  String legend "Legend";
end Curve;
\end{lstlisting}

The mandatory \lstinline!x! and \lstinline!y! expressions are restricted to be
component references refering to a scalar variable or \lstinline!time!. It
is an error if \lstinline!x! or \lstinline!y! does not designate a scalar variable.
It is an error if the unit of an expression (i.e., a variable's \lstinline!unit!,
or second for \lstinline!time!) is incompatible with the \lstinline!unit! of the
corresponding axis.

When \lstinline!legend! isn't provided, the tool produces a default based on
\lstinline!x! and/or \lstinline!y!.  Providing the empty string as
\lstinline!legend! means that the curve shall be omitted from the plot legend.

> Variable replacements, as described in \autoref{variable-replacements}, can be

Reviewer: This should be `\cref` instead of `\autoref` after the merge with _master_.

> Variable replacements, as described in \autoref{variable-replacements}, can be

Reviewer: ```suggestion
Variable replacements, as described in \cref{variable-replacements}, can be
```

Author: Thanks, I must have missed the when I merged (to large conflict...)

When an axis bound isn't provided, the tool computes one automatically.

The Modelica tool is responsible for showing the unit used for values at the
axis tick marks, so the axis \lstinline!label! shall not contain the unit.

If the specified unit is not supported by the tool, or the unit does not
represent a display unit of the plotted variable (for example,

> \lstinline!unit="cm"! and variable denotes a mass), the behavior is unspecified.

Reviewer: Again, the use of _unspecified_ opens up for various tools giving this a nice treatment, which will result in figures developed with different tools to become non-portable without a clear statement in the specification saying that the figures are incorrect.

Reviewer: Looks good now.
